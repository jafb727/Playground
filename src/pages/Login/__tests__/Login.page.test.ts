/**
 * @author: Jose A Felix
 * @editor: Jose A Felix
 * @name: Login.page.test
 * @description: Login page test cases
 */

/* --------------------------------------------- */

/** @import Libraries */
import { act, screen } from "@testing-library/react";

/** @import Components */
import Component from "../Login.page";

/** @import Utilities */
import { basicSetup } from "../../../test/setupTests";

/* --------------------------------------------- */

describe("Login.page", () => {
   test("Renders component ok", async () => {
      basicSetup(Component);
      const component = screen.getByTestId("login");

      await act(() => {
         expect(component).toBeInTheDocument();
      });
   });

   /* ----------------------- */

   test("Passes shallow snapshot test", async () => {
      const component = basicSetup(Component);

      await act(() => {
         expect(component).toMatchSnapshot();
      });
   });
});
