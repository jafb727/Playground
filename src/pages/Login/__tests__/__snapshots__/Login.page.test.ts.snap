// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.page Passes shallow snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
        data-testid="login"
        id="login"
      >
        <div
          class="defaultBackground undefined MuiBox-root css-0"
          data-testid="background"
        >
          <div
            class="full-size-container MuiBox-root css-0"
            id="workspace"
          >
            <header
              class="header-full-page"
              id="header"
            >
              <div
                class="ribbonContainer ribbon MuiBox-root css-1fn0841"
                data-testid="ribbon"
              >
                <div
                  class="MuiBox-root css-0"
                  data-testid="logo"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4 logo css-95oyd8-MuiTypography-root"
                  >
                    MY
                  </h4>
                </div>
              </div>
            </header>
            <section
              class="full-size-container"
              id="content"
            />
            <footer
              class="footer-full-page"
              id="footer"
            >
              <div
                class="ribbonContainer ribbon MuiBox-root css-1a9getn"
                data-testid="ribbon"
              >
                <div
                  class="MuiBox-root css-0"
                  data-testid="copyright-component"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-n6q3br-MuiTypography-root"
                  >
                    © 
                    All rights reserved.
                  </p>
                </div>
              </div>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
      data-testid="login"
      id="login"
    >
      <div
        class="defaultBackground undefined MuiBox-root css-0"
        data-testid="background"
      >
        <div
          class="full-size-container MuiBox-root css-0"
          id="workspace"
        >
          <header
            class="header-full-page"
            id="header"
          >
            <div
              class="ribbonContainer ribbon MuiBox-root css-1fn0841"
              data-testid="ribbon"
            >
              <div
                class="MuiBox-root css-0"
                data-testid="logo"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h4 logo css-95oyd8-MuiTypography-root"
                >
                  MY
                </h4>
              </div>
            </div>
          </header>
          <section
            class="full-size-container"
            id="content"
          />
          <footer
            class="footer-full-page"
            id="footer"
          >
            <div
              class="ribbonContainer ribbon MuiBox-root css-1a9getn"
              data-testid="ribbon"
            >
              <div
                class="MuiBox-root css-0"
                data-testid="copyright-component"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-n6q3br-MuiTypography-root"
                >
                  © 
                  All rights reserved.
                </p>
              </div>
            </div>
          </footer>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
